### PR TITLE
[vcs] expand commits apis

### PR DESCRIPTION
**Pull request by dcramer**
- add release commits list endpoint
- support creating commits via attribute to webhook (and thus release details)
- make reposting of release details acceptable

### CODE REVIEW

File: src/sentry/api/endpoints/project_releases.py [hunk 1]
```diff
@@ -131,29 +143,45 @@ def post(self, request, project):
...
+                release, created = Release.objects.get(
+                    project=project,
+                    version=result['version'],
+                ), False
+
+            commit_list = result.get('commits')
+            if commit_list:
+                hook = ReleaseHook(project)
+                # TODO(dcramer): handle errors with release payloads
+                hook.set_commits(release.version, result['commits'])
```
Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_releases.py]: Might as well just use `commit_list` here

File: src/sentry/api/endpoints/project_releases.py [hunk 2]
```diff
@@ -24,6 +25,7 @@ def create_new_release_scenario(runner):
         data={
             'version': '2.0rc2',
             'ref': '6ba09a7c53235ee8a8fa5ee4c1ca8ca886e7fdbb',
+            # 'commits': [{'id': 'a' * 40}, {'id': 'b' * 40}],
```
Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_releases.py]: Remove?

**Review by mattrobenolt — APPROVED** [review]

**Comment by mattrobenolt** [comment]
Prob worth of CHANGES too.

File: src/sentry/api/endpoints/project_releases.py [hunk 3]
```diff
@@ -37,13 +39,23 @@ def list_releases_scenario(runner):
     )
 
 
+class CommitSerializer(serializers.Serializer):
+    id = serializers.CharField(max_length=64)
+    repository = serializers.CharField(max_length=64, required=False)
```
Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_releases.py]: Hmm, why isn't this required?

Comment (dcramer) [inline src/sentry/api/endpoints/project_releases.py]: we simplify the default repository flow -- if you dont specify it all commits get lumped into one repo bound to the project name

**Comment by dcramer** [comment]
Added support for PUT too per convo of requirement

**Pull request merged**